### PR TITLE
fix: mm connect flow

### PR DESCRIPTION
**Pull request by itaven**
- updated mm `connectWallet()` flow 
- updated locator/flow  for `confirmTransactionOfTokenApproval()`
- added viem for `wallets` package to get  `getCheckSum` address 0x107209ce35ab1a0b3a22593ba85382817f471cd -> 0x107209CE35Ab1a0B3a22593ba85382817F471CDe since MM shows only in lower case. Bumped ts version to resolve conflicts with viem.
   - MM provide checkSummed address but only with "Copy to clipboard" button but for wallets setup there is no option to  get from clippoard since [metamask/lavamoat](https://github.com/LavaMoat/LavaMoat/pull/360) blocked it.

### CODE REVIEW

File: packages/wallets/src/metamask/metamask.page.ts [hunk 1]
```diff
@@ -240,10 +240,13 @@ export class MetamaskPage implements WalletPage {
       await this.navigate();
       await this.header.optionsMenuButton.click();
       await this.optionsMenu.menuAccountDetailsButton.click();
-      const address =
-        await this.popoverElements.accountDetailCopyAddressButton.textContent();
+      const address = await this.page
+        .locator(
+          '//div[@data-testid="address-copy-button-text"]/preceding-sibling::p',
```
Comment (jake4take) [inline packages/wallets/src/metamask/metamask.page.ts:245]: can you create a new locator "address" in that class?

Comment (itaven) [reply packages/wallets/src/metamask/metamask.page.ts:245]: sure, my bad

Comment (itaven) [reply packages/wallets/src/metamask/metamask.page.ts:245]: I'll stay with popover elements for now

Comment (itaven) [reply packages/wallets/src/metamask/metamask.page.ts]: Fixed

**Review by jake4take — DISMISSED** [review]

**Review by Vorobeyko — APPROVED** [review]

**Pull request merged**